### ISSUE TITLE
Question about labelling strategy

### ISSUE THREAD

**Issue by DarylWM** [AlexeyAB/darknet#5762]
Hi @AlexeyAB. I have a custom data set of mass spectrometry data converted to 90,000 images where not all the objects I want to detect are labelled. There are three classes labelled: 180K instances of class 0, 23K of class 1, and 1.5K of class 2.

80% of the images are selected at random and assigned to the training set; the remainder are assigned to the validation set. From the training set I'm removing the unlabelled objects by setting them to the same colour as the background. I understand the validation set should have real-world images so I'm not altering those images.

During training, average loss is quite low (less than 0.5) but mAP is only about 1-2%. In answer to some other questions you recommend making the training set more representative, and I can work on more class examples and also the class imbalance.

Am I right that only the training set should be manipulated, in this case by removing unlabelled objects, and the validation set should be unaltered real data?

**Comment by pshwetank** [comment]
You must put the negative samples(objects that you dont want to detect) as blank in the coordinate file. 
So, suppose, you dont want to detect a specific class, say class 3. So, try to collect the images that only have class 3 and nothing else. Now, you might have an image from the set of such images, lets say its image1_class3.jpg. Then, keep its label file image1_class3.txt as blank. The label files of such images should not have any label. 
This way, your classifier will be able to understand what to ignore in the image data.
